### PR TITLE
Support Multiple Servers

### PR DESCRIPTION
**Pull request by brigand**
This adds support for multiple servers (so the bot can also connect to [Libera](https://libera.chat/)). Config is now a list of servers, and a shared plugin config. It at least basically works, when running it locally.

Need another pair of eyes on whether anything is/isn't shared between the connections that shouldn't/should be.

cc @ljharb @caub 

(Will fix the tests before merging)

### CODE REVIEW

**Comment by ljharb** [comment]
Sans review, I’d expect factoids to be live-shared, but |> commands to be per-network.

File: src/bot.js [hunk 1]
```diff
@@ -11,153 +11,175 @@ const changeNick = require('./utils/changeNick');
...
-      client.say(isRoom ? to : from, `Internal error while processing the message`);
+    let messageObj;
+    try {
+      // eslint-disable-next-line global-require
+      const processMessage = require('./processMessage');
+      messageObj = processMessage(client, config, logs, from, to, message);
+    } catch (e) {
+      const isRoom = /^#/.test(to);
+      if (Date.now() > lastProcessMessageFail + 1000 * 60 * 60) {
+        lastProcessMessageFail = Date.now();
```
Comment (ljharb) [inline src/bot.js:100]:
```suggestion
      const now = Date.now();
      if (now > lastProcessMessageFail + 1000 * 60 * 60) {
        lastProcessMessageFail = now;
```

**Review by ljharb — APPROVED** [review]
seems good?

**Comment by caub** [comment]
As a global picture, before jellobot would only listen to freenode request, and now with this PR, it'll serve mutiple kind of irc servers? nice, you just need to create an account for jellobot on each of those servers right?

the code looks good to me, with what ljharb reviewed

**Comment by brigand** [comment]
FYI, also had to switch from the irc package to the [irc-upd](https://www.npmjs.com/package/irc-upd) fork, as the former has a broken SASL implementation, which means it can't connect to Libera from a cloud server.

https://github.com/brigand/jellobot/commit/c90c040f459ed74c838cdc2020d0121814c9d225

**Pull request merged**